src/ProgressBar/index.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useLayoutEffect, useRef, useState } from "react";
+
+interface ProgressBarProps {
+  /**
+   * Int from 0 to 100
+   */
+  percentComplete: number;
+  /**
+   * CSS Value; Total width of progress bar.
+   * (e.g. "50%" or "200px")
+   */
+  totalWidth?: string;
+  /**
+   * Size (height) of the ProgressBar
+   */
+  size?: "s" | "m";
+}
+
+const ProgressBar = ({
+  totalWidth = "100%",
+  percentComplete = 0,
+  size = "m",
+}: ProgressBarProps) => {
+  const [currentPct, setCurrentPct] = useState(0);
+  const lineRef = useRef<SVGLineElement>(null);
+  const animateRef = useRef<SVGAnimateElement>(null);
+
+  useLayoutEffect(() => {
+    const animFrom = percentComplete === 0 ? percentComplete : currentPct;
+    if (lineRef.current) {
+      if (animateRef.current) {
+        animateRef.current.setAttribute("from", `${animFrom}%`);
+        animateRef.current.setAttribute("to", `${percentComplete}%`);
+        animateRef.current.beginElement();
+        setCurrentPct(percentComplete);
+      }
+    }
+  }, [percentComplete]);
+
+  return (
+    <div
+      className={`nds-progressbar nds-progressbar--${size}`}
+      style={{ width: `${totalWidth}` }}
+    >
+      {percentComplete > 0 && (
```

Comment:
The rounded line cap in SVG is drawn from the _end_ of the line, so with an `x2` value of `0`, I was seeing a little round thing peek out. This prevents the progress line from being drawn if there's nothing to draw.